crates/autopilot/src/domain/settlement/mod.rs

```diff
@@ -140,6 +141,75 @@ impl Settlement {
 
         Ok(Self { trades, auction_id })
     }
+
+    /// Build a settlement from a `competition::Solution`.
+    ///
+    /// Since `competition::Solution` does not contain JIT orders, the resulting
+    /// settlement might be a subset of it's onchain observable twin.
+    pub fn from_solution(
+        solution: &competition::Solution,
+        auction: &auction::Auction,
+        auction_id: auction::Id,
+    ) -> Result<Self, error::Solution> {
+        let auction_orders = auction
+            .orders
+            .iter()
+            .map(|o| (o.uid, o))
+            .collect::<HashMap<_, _>>();
+
+        let mut trades = Vec::with_capacity(solution.orders().len());
+        for (order_uid, traded) in solution.orders().iter() {
+            let order = auction_orders
+                .get(order_uid)
+                .ok_or(error::Solution::MissingOrder)?;
```

Comment:
This get's hit if driver reports JIT order in `/solve` request. Or it reports an outside of auction order.

Reply:
So if this can happen normally with JIT orders there is no reason to error out here, right?

Reply:
It depends if we want to ignore the fact that the driver included JIT orders in the `/solve` response. Right now they don't include them, so I would expect this to continue as an API rule.

However, if there would be a need in the future, we could improve this. During competition time, for all orders from `/solve` response, we would check if they exist in the database, and determine if they are JIT or out-of-auction.
Note: this improved way ☝️ will be implemented in postprocessing time, where we have to be certain 100%.